yateto/codegen/visitor.py

```diff
@@ -599,7 +643,26 @@ def iterate_collect(self):
       cur_dict[base_name, base_name_without_ns] = tensors
     # Don't forget last namespace
     yield cur_namespace, cur_dict
-    
+
+  def iterate_collect_scalar(self):
+    cur_namespace = ''
+    cur_dict = collections.OrderedDict()
+    cur_dictScalar = collections.OrderedDict()
+    for base_name, tensors in self._scalarCollect.items():
+      splitted = base_name.rsplit('::', 1)
```

Comment:
split

Reply:
`rsplit` should be correct here: e.g. `a::b::c::krnl` should result in the kernel name `krnl` and the namespace `a::b::c`

Reply:
Sorry, what I meant: 'splitted' isn't a word...

Reply:
Ok, renamed it.

Reply:
(even though there is still blatant code duplication w.r.t. the method above it)